### PR TITLE
snap: fix problem with read-only fs (#232)

### LINKED ISSUE #232: renku update failing with Errno 30
Running renku update gives me the following error, related to read-only. I'm working on ubuntu 18.04 LTS.  
```
Traceback (most recent call last):
  File "/snap/renku/x1/bin/renku", line 11, in <module>
    load_entry_point('renku==0.1.0.dev20180628', 'console_scripts', 'renku')()
  File "/snap/renku/x1/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/snap/renku/x1/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/snap/renku/x1/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/snap/renku/x1/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/snap/renku/x1/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/snap/renku/x1/lib/python3.5/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/snap/renku/x1/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/snap/renku/x1/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/snap/renku/x1/usr/lib/python3.5/contextlib.py", line 30, in inner
    return func(*args, **kwds)
  File "/snap/renku/x1/lib/python3.5/site-packages/renku/cli/update.py", line 158, in update
    execute(client, output_file, output_paths=output_paths)
  File "/snap/renku/x1/lib/python3.5/site-packages/renku/cli/_cwl.py", line 32, in execute
    import cwltool.factory
  File "/snap/renku/x1/lib/python3.5/site-packages/cwltool/factory.py", line 6, in <module>
    from . import load_tool, workflow
  File "/snap/renku/x1/lib/python3.5/site-packages/cwltool/load_tool.py", line 20, in <module>
    import schema_salad.schema as schema
  File "/snap/renku/x1/lib/python3.5/site-packages/schema_salad/__init__.py", line 38, in <module>
    import avro
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "/snap/renku/x1/lib/python3.5/site-packages/past/translation/__init__.py", line 406, in load_module
    os.makedirs(dirname)
  File "/snap/renku/x1/usr/lib/python3.5/os.py", line 241, in makedirs
    mkdir(name, mode)
OSError: [Errno 30] Read-only file system: '/snap/renku/x1/lib/python3.5/site-packages/avro/__pycache__'
```

### PR DESCRIPTION
**Pull request by leafty**
This commit shows how to fix issue #232 by patching future.
For some reason, I can't set the version to be something like 'future==0.16.0patch',
snapcraft fails when trying to install it.

You can check the patch here: https://github.com/PythonCharmers/python-future/compare/v0.16.x...leafty:fix-read-only?expand=1

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -37,3 +37,4 @@ parts:
     python-version: python3
     source: .
     source-type: git
+    process-dependency-links: true
```
Comment (jirikuncar) [inline snap/snapcraft.yaml]: can you move the `dependency_links` from `setup.py` to `python-packages:` here?

File: setup.py [hunk 2]
```diff
@@ -85,6 +85,7 @@
     'requests>=2.18.4',
     'tabulate>=0.7.7',
     'werkzeug>=0.12',
+    'future==0.16.0'
```
Comment (jirikuncar) [inline setup.py]: IMHO it should not be necessary for non-snap packages.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (leafty) [reply snap/snapcraft.yaml]: I'll try. Didn't see it was an option.

Comment (jirikuncar) [reply snap/snapcraft.yaml]: Do we still need `process-dependency-links: true`?

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -37,3 +37,6 @@ parts:
...
+    python-packages:
+      - "git+https://github.com/leafty/python-future.git@fix-read-only#egg=future-0.16.0"
```
Comment (jirikuncar) [inline snap/snapcraft.yaml]: 1. Will you submit a PR to the upstream project?
2. Can you move the changes to `https://github.com/SwissDataScienceCenter/python-future.git`?

Comment (leafty) [reply snap/snapcraft.yaml]: 1. Yes, wanted to know first if the fix works
2. Will do

**Comment by jirikuncar** [comment]
@leafty can we also change the `Confinement: classic` so we can release it in public store?

**Comment by leafty** [comment]
> @leafty can we also change the Confinement: classic so we can release it in public store?

When I package the snap in devmode, the CLI complains about not seeing git.

**Comment by leafty** [comment]
Here's the log:
```
Traceback (most recent call last):
  File "/snap/renku/x6/bin/renku", line 11, in <module>
    load_entry_point('renku==0.1.0.dev20180628', 'console_scripts', 'renku')()
  File "/snap/renku/x6/lib/python3.5/site-packages/pkg_resources/__init__.py", line 476, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/snap/renku/x6/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2700, in load_entry_point
    return ep.load()
  File "/snap/renku/x6/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2318, in load
    return self.resolve()
  File "/snap/renku/x6/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2324, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/snap/renku/x6/lib/python3.5/site-packages/renku/__init__.py", line 22, in <module>
    from .api import APIClient, LocalClient
  File "/snap/renku/x6/lib/python3.5/site-packages/renku/api/__init__.py", line 20, in <module>
    from .client import APIClient, LocalClient
  File "/snap/renku/x6/lib/python3.5/site-packages/renku/api/client.py", line 31, in <module>
    from .datasets import DatasetsApiMixin
  File "/snap/renku/x6/lib/python3.5/site-packages/renku/api/datasets.py", line 28, in <module>
    import git
  File "/snap/renku/x6/lib/python3.5/site-packages/git/__init__.py", line 82, in <module>
    refresh()
  File "/snap/renku/x6/lib/python3.5/site-packages/git/__init__.py", line 73, in refresh
    if not Git.refresh(path=path):
  File "/snap/renku/x6/lib/python3.5/site-packages/git/cmd.py", line 290, in refresh
    raise ImportError(err)
ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet
```

**Comment by leafty** [comment]
Now, I have the following issues:
```bash
This system supports the C.UTF-8 locale which is recommended.
You might be able to resolve your issue by exporting the
following environment variables:

    export LC_ALL=C.UTF-8
    export LANG=C.UTF-8

Click discovered that you exported a UTF-8 locale
but the locale system could not pick up from it because
it does not exist.  The exported locale is "en_GB.UTF-8" but it
is not supported
```

```bash
$ mkdir test
$ cd test
$ LC_ALL=C.UTF-8 renku init
Traceback (most recent call last):
  File "/snap/renku/x7/bin/renku", line 11, in <module>
    load_entry_point('renku==0.1.0.dev20180628', 'console_scripts', 'renku')()
  File "/snap/renku/x7/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/snap/renku/x7/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/snap/renku/x7/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/snap/renku/x7/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/snap/renku/x7/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/snap/renku/x7/lib/python3.5/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/snap/renku/x7/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/snap/renku/x7/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/snap/renku/x7/usr/lib/python3.5/contextlib.py", line 30, in inner
    return func(*args, **kwds)
  File "/snap/renku/x7/lib/python3.5/site-packages/renku/cli/init.py", line 112, in init
    name=name, force=force, use_external_storage=use_external_storage
  File "/snap/renku/x7/lib/python3.5/site-packages/renku/api/repository.py", line 162, in init_repository
    self.renku_path.mkdir(parents=True, exist_ok=force)
  File "/snap/renku/x7/usr/lib/python3.5/pathlib.py", line 1220, in mkdir
    self._accessor.mkdir(self, mode)
  File "/snap/renku/x7/usr/lib/python3.5/pathlib.py", line 371, in wrapped
    return strfunc(str(pathobj), *args)
PermissionError: [Errno 13] Permission denied: '/home/johann/test/.renku'
```

**Pull request merged**